plugins/woocommerce/client/blocks/assets/js/blocks/product-collection/edit/inspector-controls/use-carousel-layout-adjustments.ts

```diff
@@ -0,0 +1,162 @@
+/**
+ * External dependencies
+ */
+import { useRef, useEffect } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { createBlock, BlockInstance } from '@wordpress/blocks';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import {
+	coreQueryPaginationBlockName,
+	productTemplateBlockName,
+	nextPreviousArrowsBlockName,
+} from '../../constants';
+import { LayoutOptions, type ProductCollectionAttributes } from '../../types';
+
+/**
+ * Custom hook to adjust the pagination block when switching between layouts.
+ *
+ * @param {string} clientId - The client ID of the product collection block.
+ * @param {ProductCollectionAttributes} attributes - The attributes of the product collection block.
+ * @return {void}
+ */
+const useLayoutAdjustments = (
+	clientId: string,
+	attributes: ProductCollectionAttributes
+) => {
+	const { displayLayout, collection } = attributes;
+
+	const previousLayoutType = useRef< LayoutOptions >( displayLayout.type );
+	const innerBlocks = useSelect(
+		( select ) =>
+			clientId ? select( blockEditorStore ).getBlocks( clientId ) : [],
+		[ clientId ]
+	);
+
+	const productTemplateBlocks = innerBlocks.filter(
+		( block: BlockInstance ) => block.name === productTemplateBlockName
+	);
+	const productTemplateBlock = productTemplateBlocks[ 0 ];
+	const productTemplateBlockClientId = productTemplateBlock?.clientId;
+
+	const productTemplateBlockIndex = useSelect(
+		( select ) =>
+			productTemplateBlockClientId
+				? select( blockEditorStore ).getBlockIndex(
+						productTemplateBlockClientId
+				  )
+				: 0,
+		[ productTemplateBlockClientId ]
+	);
+
+	const { insertBlock, removeBlock, replaceBlock } =
+		useDispatch( blockEditorStore );
+
+	useEffect( () => {
```

Comment:
What do you think to create two different functions `handleCarouselTransition` and `handleNonCarouselTransition` functions and use those in the `useEffect`? I think that it could increase the readibility

Reply:
Done in https://github.com/woocommerce/woocommerce/pull/59519/commits/eeb1d12cd064b80a4f5eb3942e6f809aa0421281